crates/shared/src/price_estimation/native_price_cache.rs

```diff
@@ -193,6 +194,19 @@ impl Inner {
     }
 }
 
+fn should_cache(result: &Result<f64, PriceEstimationError>) -> bool {
+    // We don't want to cache errors that we consider transient
+    match result {
+        Ok(_) => true,
+        Err(PriceEstimationError::NoLiquidity) => true,
+        Err(PriceEstimationError::ZeroAmount) => true,
```

Comment:
Marking `ZeroAmount` as cache worthy probably sends a wrong message to the reader. This case should never happen so I would consider it a bug in our system so caching that should not be an option IMO.